### PR TITLE
test_cephfs_shell: use StringIO instead of BytesIO

### PR DESCRIPTION
**Pull request by rishabh-d-dave**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by rishabh-d-dave** [comment]
Tests were successful; results are here - https://github.com/ceph/ceph/pull/32288#issuecomment-633851091. Originally this commit resided on that PR but I'm moving it on a separate PR so that this can merged faster.

**Comment by varshar16** [comment]
@rishabh-d-dave Why is this change required?

**Comment by rishabh-d-dave** [comment]
Our qa/ code uses both StringIO and BytesIO which makes it slightly confusing. So I'm replacing for uniformity. I've added that to the commit message.

**Comment by batrick** [comment]
@kshtsk can you please add your 2 cents.

Also, @kshtsk can you add your real name to your github profile so it's easier to search for you in the reviewer request drop down.

**Comment by kshtsk** [comment]
> @kshtsk can you please add your 2 cents.
> 
> Also, @kshtsk can you add your real name to your github profile so it's easier to search for you in the reviewer request drop down.

actually I don't see any advantage of adding this commit, however it should not break anything, in general, there is a lot of code duplication, for example, `get_cephfs_shell_cmd_output` in combination with `run_cephfs_shell_cmd` it exactly the same as `remote.sh()` does, which makes `run_cephfs_shell_cmd` useless. Next, `get_cephfs_shell_cmd_error` results can be achieved with
```
errorstream = StringIO()
remote.sh(cmd, stderr=errors)
```
I prefer to use BinaryIO by defaults since it will allow you to process binary output, or data without extra converstion to unicode or so.

Finally, I would remove `python3` label from this PR since the script is already python3 ready, and it looks  not providing anything new that python3 specific.

I'm not sure if this is what you expected to hear from me. Anyways I would not block this PR.

**Comment by rishabh-d-dave** [comment]
> general, there is a lot of code duplication, for example, get_cephfs_shell_cmd_output in combination with run_cephfs_shell_cmd it exactly the same as remote.sh() does, which makes run_cephfs_shell_cmd useless.

`run_cephfs_shell_cmd()` also prepends `cephfs -c <path/to/conf> --` to every command it receives. So the combination is not exactly the same. 

Using both StringIO and BytesIO with actual necessity can lead to confusion and unnecessary bugs or test failures. StringIO is more frequently used and is easier to use for tests. So the intent of this PR is just to replace with StringIO uniformity's sake.
```
$ grep -r BytesIO qa/tasks/cephfs | wc -l
18
$ grep -r StringIO qa/tasks/cephfs | wc -l
31
```

File: qa/tasks/cephfs/test_cephfs_shell.py [hunk 1]
```diff
@@ -50,8 +50,8 @@ def run_cephfs_shell_cmd(self, cmd, mount_x=None, shell_conf_path=None,
         args.extend(("--", cmd))
 
         log.info("Running command: {}".format(" ".join(args)))
-        return mount_x.client_remote.run(args=args, stdout=BytesIO(),
-                                         stderr=BytesIO(), stdin=stdin)
+        return mount_x.client_remote.run(args=args, stdout=StringIO(),
+                                         stderr=StringIO(), stdin=stdin)
```
Comment (batrick) [inline qa/tasks/cephfs/test_cephfs_shell.py:54]: Here's what I suggest as a compromise: make it caller defined (with a reasonable default). So instead:

````
def run_cephfs_shell_cmd(self, cmd, mount_x=None, shell_conf_path=None, opts=None, stdin=None, stdout=StringIO(), stderr = StringIO()):
    ...
    return mount_x.client_remote.run(args=args, stdout=stdout, stderr=stderr, stdin=stdin)
````

Ditto for `run_cephfs_shell_script`.

Comment (rishabh-d-dave) [reply qa/tasks/cephfs/test_cephfs_shell.py]: Better yet, IMO. Made the change.

File: qa/tasks/cephfs/test_cephfs_shell.py [hunk 2]
```diff
@@ -75,8 +76,8 @@ def get_cephfs_shell_script_output(self, script, mount_x=None,
             opts=opts, stdin=stdin).stdout.getvalue().strip())
 
     def run_cephfs_shell_script(self, script, mount_x=None,
-                                shell_conf_path=None, opts=None,
-                                stdin=None):
+                                shell_conf_path=None, opts=None, stdin=None,
+                                stdout=StringIO(), stderr=String()):
```
Comment (batrick) [inline qa/tasks/cephfs/test_cephfs_shell.py:80]: `StringIO()`?

**Comment by tchaikov** [comment]
@batrick why did you remove the python3 label? i am using it for tracking all python3 related change so i can backport it to previous branch if necessary. i am not sure anybody else is also using it.

**Comment by batrick** [comment]
@tchaikov 

> Finally, I would remove python3 label from this PR since the script is already python3 ready, and it looks not providing anything new that python3 specific.

https://github.com/ceph/ceph/pull/35263#issuecomment-634789439

**Comment by rishabh-d-dave** [comment]
Tests were successful - http://pulpito.ceph.com/rishabh-2020-06-04_18:33:37-fs-wip-rishabh-fs-test_cephfs_shell-fix-distro-basic-smithi/

**Comment by rishabh-d-dave** [comment]
@batrick PTAL.

**Pull request merged**